pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -307,6 +339,24 @@ func createTargetConfigController(ctx context.Context, syncCtx factory.SyncConte
 	return false, nil
 }
 
+func isExternalCloudProviderApplied(operatorSpec *operatorv1.StaticPodOperatorSpec) (bool, error) {
+	var observedConfig map[string]interface{}
+	if err := yaml.Unmarshal(operatorSpec.ObservedConfig.Raw, &observedConfig); err != nil {
+		return false, fmt.Errorf("failed to unmarshal the observedConfig: %v", err)
+	}
+
+	cloudProvider, found, err := unstructured.NestedSlice(observedConfig, "extendedArguments", "cloud-provider")
+	if !found || err != nil {
+		return false, err
+	}
+
+	if len(cloudProvider) == 1 && (cloudProvider[0] == "external" || cloudProvider[0] == "") {
+		return true, nil
+	}
```

Comment:
`cloud-provider == ""` is not the same as being external. Does this imply that on a platform none or other platform that doesn't currently have in-tree, we are going to add the condition to say KCM doesn't own a cloud controller, even though no cloud controller exists?

At first glance I can't see any harm in this, just wasn't expecting that 🤔